Dockerfile

luandro opened the pull request:
Added Dockerfile and updated README to include how to use it.

dangowrt committed:
ar71xx: add support for LibreRouter v1

Based on 6325eeec2d72a4b14883c46b8423ba8580cbd2e6 by Daniel Golle

dangowrt committed:
ar71xx: add support for LibreRouter v1

Based on 6325eeec2d72a4b14883c46b8423ba8580cbd2e6 by Daniel Golle

dangowrt committed:
ar71xx: war: spi-nor: force-disable 4-byte mode

only support flash chips up to 16mb and truncate larger chips
instead of attempting to switch to 4-byte addressing mode which
doesn't work on QCA SPI hardware nor the rom loader.

dangowrt committed:
ar71xx: war: spi-nor: force-disable 4-byte mode

only support flash chips up to 16mb and truncate larger chips
instead of attempting to switch to 4-byte addressing mode which
doesn't work on QCA SPI hardware nor the rom loader.

diizzyy committed:
tools/bison: Update to 3.0.5

Update bison to 3.0.5
Bugfix release
Remove 001-fix-macos-vasnprintf.patch as it is fixed upstream

Signed-off-by: Daniel Engberg <daniel.engberg.lists@pyret.net>
(cherry picked from commit df02e7a3c790552c9620242544ec0137dae6a32b)

robimarko committed:
firmware-utils: tplink-safeloader: Add CPE210 v3

Add TP-Link CPE210 v3 to the support list.
Its identical to the v2.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
ar71xx: Add support for TP-Link CPE210 v3

 ar71xx: Add support for TP-Link CPE210 v3.
Looks identical to the v2.

This PR adds support for a popular low-cost 2.4GHz N based AP

Specifications:
 - SoC: Qualcomm Atheros QCA9533 (650MHz)
 - RAM: 64MB
 - Storage: 8 MB SPI NOR
 - Wireless: 2.4GHz N based built into SoC 2x2
 - Ethernet: 1x 100/10 Mbps, integrated into SoC, 24V POE IN

Installation:
Flash factory image through stock firmware WEB UI
or through TFTP
To get to TFTP recovery just hold reset button while powering on for
around 4-5 seconds and release.
Rename factory image to recovery.bin
Stock TFTP server IP:192.168.0.100
Stock device TFTP adress:192.168.0.254

Signed-off-by: Robert Marko <robimarko@gmail.com>

spiccinini committed:
LibreRouter: use cmdline from u-boot

spiccinini committed:
LibreRouter: use cmdline from u-boot

spiccinini committed:
ar71xx: add secondary QCA9558 UART (UART1) to LibreRouter

spiccinini committed:
ar71xx: add secondary QCA9558 UART (UART1) to LibreRouter

spiccinini committed:
select MACH_LIBREROUTERV1 by default

spiccinini committed:
select MACH_LIBREROUTERV1 by default

spiccinini committed:
add .config and feeds to build LibreMesh firmware

This is a only a starting point.

spiccinini committed:
add .config and feeds to build LibreMesh firmware

This is a only a starting point.

spiccinini committed:
Disable CONFIG_DOWNLOAD_FOLDER by default

CONFIG_DOWNLOAD_FOLDER is a very handy way of caching and sharing
the source download. We should find a way of leting the user enable this
and selecting the folder.

spiccinini committed:
Disable CONFIG_DOWNLOAD_FOLDER by default

CONFIG_DOWNLOAD_FOLDER is a very handy way of caching and sharing
the source download. We should find a way of leting the user enable this
and selecting the folder.

Commit:
add new wireless regulation specs for Vatican and Tanzania

Commit:
add new wireless regulation specs for Vatican and Tanzania

spiccinini committed:
add build instructions in README.md

spiccinini committed:
add build instructions in README.md

spiccinini committed:
uboot-envtools: add support for LibreRouter v1

Signed-off-by: Santiago Piccinini <spiccinini@altermundi.net>

spiccinini committed:
uboot-envtools: add support for LibreRouter v1

Signed-off-by: Santiago Piccinini <spiccinini@altermundi.net>

spiccinini committed:
Update issue templates

spiccinini committed:
Update issue templates

spiccinini committed:
remove commented CONFIG_DOWNLOAD_FOLDER

This option is not used but generates confusion as it is different
than the default diffconfig

spiccinini committed:
remove commented CONFIG_DOWNLOAD_FOLDER

This option is not used but generates confusion as it is different
than the default diffconfig

spiccinini committed:
enable fft-eval from lime-packages master

spiccinini committed:
enable fft-eval from lime-packages master

spiccinini committed:
deselect lime meta packages

spiccinini committed:
deselect lime meta packages

spiccinini committed:
do not select bm6 by default

spiccinini committed:
do not select bm6 by default

spiccinini committed:
deselect lime-webui

spiccinini committed:
deselect lime-webui

spiccinini committed:
disable all CONFIG_FEED_* for the moment

spiccinini committed:
disable all CONFIG_FEED_* for the moment

spiccinini committed:
Merge remote-tracking branch 'librerouterorg/librerouter-18.06' into librerouter-18.06

spiccinini committed:
deselect lime-map-agent

spiccinini committed:
deselect lime-map-agent

spiccinini committed:
remove dnsmasq from DEFAULT_PACKAGES

spiccinini committed:
remove dnsmasq from DEFAULT_PACKAGES

spiccinini committed:
deselect luci admin full

spiccinini committed:
deselect luci admin full

spiccinini committed:
add DEVICE_TYPPE librerouter and remove breaking dependencies

spiccinini committed:
add DEVICE_TYPPE librerouter and remove breaking dependencies

spiccinini committed:
add kmod-usb-storage and safe-upgade device dependencies

spiccinini committed:
add kmod-usb-storage and safe-upgade device dependencies

spiccinini committed:
use libremap-gent from libremesh github

spiccinini committed:
use libremap-gent from libremesh github

spiccinini committed:
select babeld

spiccinini committed:
select babeld

spiccinini committed:
add kernel reset button support

spiccinini committed:
add kernel reset button support

spiccinini committed:
enable CONFIG_PACKAGE_ALFRED_BATHOSTS

spiccinini committed:
enable CONFIG_PACKAGE_ALFRED_BATHOSTS

spiccinini committed:
enable busybox CROND_SPECIAL_TIMES @reboot, @hourly, etc

spiccinini committed:
enable busybox CROND_SPECIAL_TIMES @reboot, @hourly, etc

spiccinini committed:
use routing fork to test babeld-1.9-dev

spiccinini committed:
use routing fork to test babeld-1.9-dev

spiccinini committed:
ar71xx: fix librerouter switch config

Not both ethernet ports (GMAC0/RGMII and GMAC1/SGMII) are working

spiccinini committed:
ar71xx: fix librerouter switch config

Not both ethernet ports (GMAC0/RGMII and GMAC1/SGMII) are working

spiccinini committed:
fix 'use routing fork to test babeld-1.9-dev' hash

spiccinini committed:
fix 'use routing fork to test babeld-1.9-dev' hash

spiccinini committed:
update first-boot-wizard package name

spiccinini committed:
update first-boot-wizard package name

spiccinini committed:
select shared-state

spiccinini committed:
select shared-state

spiccinini committed:
deselected smonit and dnsmasq-distributed-hosts

spiccinini committed:
deselected smonit and dnsmasq-distributed-hosts

spiccinini committed:
default config with all partitions R/W

Allow R/W to all partitions to ease migrations of bootloaders,
changing MAC addresses, etc.
This config is overrided by the bootloader's bootargs mtdparts

spiccinini committed:
default config with all partitions R/W

Allow R/W to all partitions to ease migrations of bootloaders,
changing MAC addresses, etc.
This config is overrided by the bootloader's bootargs mtdparts

spiccinini committed:
support librerouter v0 and v1 boards with v1 firmware

spiccinini committed:
support librerouter v0 and v1 boards with v1 firmware

spiccinini committed:
selected package ubus-lime-fbw

spiccinini committed:
selected package ubus-lime-fbw

spiccinini committed:
select uhttpd-mod-lua

spiccinini committed:
select uhttpd-mod-lua

spiccinini committed:
add lime-defaults-factory to overwrite the package file

spiccinini committed:
add lime-defaults-factory to overwrite the package file

spiccinini committed:
change default 2.4GHz mode to AP only and 5.8GHz to mesh only

The idea is that the backbone is only in 5.8GHz (at least for dual-band
routers) and 2.4GHz is not part of the mesh  providing local conectivity
near to the node.

spiccinini committed:
change default 2.4GHz mode to AP only and 5.8GHz to mesh only

The idea is that the backbone is only in 5.8GHz (at least for dual-band
routers) and 2.4GHz is not part of the mesh  providing local conectivity
near to the node.

spiccinini committed:
remove original README as it was cousing confusion

spiccinini committed:
remove original README as it was cousing confusion

spiccinini committed:
Add development instructions

spiccinini committed:
Add development instructions

spiccinini committed:
added docs notes

spiccinini committed:
added docs notes

spiccinini committed:
add more docs to the readme

spiccinini committed:
add more docs to the readme

spiccinini committed:
fixed docs

spiccinini committed:
fixed docs

spiccinini committed:
remove fbw feed

Remove as it is a libremesh package

spiccinini committed:
remove fbw feed

Remove as it is a libremesh package

spiccinini committed:
selected pirania-app, shared-state-pirania and shared-state-persist

spiccinini committed:
selected pirania-app, shared-state-pirania and shared-state-persist

luandro committed:
added dockerfile and docker readme

luandro committed:
better docker docs

spiccinini commented on README.md:
```diff
@@ -12,6 +14,13 @@ packages into package/feeds/
 
+### Building in docker container
+
+You need Docker installed, if you don't have it run `wget -O - https://get.docker.com | bash` to install.
```
I dont think it is a good idea to recomend doin wget -O | bash. Let's sugest `apt-get install docker`

luandro replied:
It's the official docker installation, `apt-get install docker` may not work for all distributions.

nicopace replied:
I would suggest to follow docker install instructions, as they may change

nicopace replied:
much more complex than years before, but these are the install docs: https://docs.docker.com/install/

spiccinini commented on README.md:
```diff
@@ -12,6 +14,13 @@ packages into package/feeds/
+You need Docker installed, if you don't have it run `wget -O - https://get.docker.com | bash` to install.
+
+Simply run `docker run -it luandro/libremesh bash` or clone this repository and create a `docker build` yourself after you've modified the necessary files. Inside the container `cd openwrt` and run `make menuconfig` to select your targets and packages. Finally run `make` to build.
+
```
How do you share the result of the work to outside of the container?
It may be a good idea to bind mount a directory from the host computer that contains the openwrt directory? So just using the container to run the binaries (make, etc) ?

luandro replied:
Good idea.

luandro replied:
I tried leaving the `openwrt` folder as a volume connected to outside the container. But found a problem that the container user needs `sudo` to execute the files. Would leaving the `output` directory outside the container be enough?

spiccinini committed:
Fix docs in readme

spiccinini committed:
Fix docs in readme

ilario committed:
default_config explicitly unset dnsmasq, useful when compiling on openwrt trunk

ilario committed:
default_config explicitly unset dnsmasq, useful when compiling on openwrt trunk

spiccinini committed:
Merge pull request #8 from ilario/recycle_for_trunk

default_config explicitly unset dnsmasq

spiccinini committed:
add minodo.info and meuno.info aliases to thisnode.info

spiccinini committed:
add minodo.info and meuno.info aliases to thisnode.info

nicopace committed:
Merge pull request #10 from LibreRouterOrg/thisnode_translation

add minodo.info and meuno.info aliases to thisnode.info

spiccinini committed:
update routing feed: "Fix babeld interface configuration in init script"

spiccinini committed:
update routing feed: "Fix babeld interface configuration in init script"

spiccinini committed:
fix 96-thisnode-info-translation sed: bad option in substitution expression

spiccinini committed:
fix 96-thisnode-info-translation sed: bad option in substitution expression

spiccinini committed:
fix uci-defaults inside /etc/config instead of /etc

spiccinini committed:
fix uci-defaults inside /etc/config instead of /etc

spiccinini committed:
change default firmware partition size to dual partition size

Change the size of the firmware partition from 16000kB to 7936kB.
This size is the same used by the bootloader. The rationale behind
this change is that if for any reason the boot process doesn't use
the correct partitions provided by the bootloader in mtdparts then
this size will be used and with a 16000kB size (only one big partiton)
default then the image will boot and jffs2 will corrupt the "stable"
partition.

spiccinini committed:
change default firmware partition size to dual partition size

Change the size of the firmware partition from 16000kB to 7936kB.
This size is the same used by the bootloader. The rationale behind
this change is that if for any reason the boot process doesn't use
the correct partitions provided by the bootloader in mtdparts then
this size will be used and with a 16000kB size (only one big partiton)
default then the image will boot and jffs2 will corrupt the "stable"
partition.

spiccinini committed:
update routing feed with babeld init fixes

spiccinini committed:
update routing feed with babeld init fixes

patogit committed:
add time as a dependency and clarify make menu process

patogit committed:
add time as a dependency and clarify make menu process

spiccinini committed:
Merge pull request #12 from patogit/patch-1

add time dependency, clarify make menu process

luandro committed:
merge latest

spiccinini committed:
http redirect / to /app by default

spiccinini committed:
http redirect / to /app by default

spiccinini committed:
remove translation of thisnode.info in uci-defaults

This doesn't work well as when the config is regenerated this won't
run again. It must be implemented in another way.

spiccinini committed:
remove translation of thisnode.info in uci-defaults

This doesn't work well as when the config is regenerated this won't
run again. It must be implemented in another way.

luandro committed:
Merge branch 'librerouter-18.06' of https://github.com/LibreRouterOrg/openwrt into librerouter-18.06

nicopace commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -1,38 +0,0 @@
----
```
this file shouldn't be removed

spiccinini committed:
select hotplug-initd-services

spiccinini committed:
select shared-state-babeld_hosts

spiccinini committed:
select missing luci-lib-libremap-babeld

spiccinini committed:
remove shared-state-date

spiccinini committed:
select check-date-http package

spiccinini committed:
deselect alfred

spiccinini committed:
add userspace watchdog support

As 18.06 does not have support of gpio-watchdog in kernel then a
'user space only' tool, om-watchdog, is used.

nicopace committed:
selects the shared-state-nodes_and_links package

this is a required package to be able to see the community map in the lime-app.

spiccinini committed:
Merge pull request #14 from LibreRouterOrg/select-nodes-and-links

selects the shared-state-nodes_and_links package

spiccinini committed:
update routing to babeld-1.9.1

spiccinini committed:
Merge pull request #15 from LibreRouterOrg/update_babeld_1.9.1

Update routing to babeld-1.9.1

spiccinini committed:
Merge pull request #16 from LibreRouterOrg/cpev3

Add cpe210-v3 support

spiccinini committed:
ar71xx: add TP-LINK CPE510 V2.0 profile

Based on the work of Andrew Cameron <apcameron@softhome.net> at:
https://github.com/aredn/aredn_ar71xx/commit/700f020e255abfd9ef8a0b2cfe9d82aeed4461da

spiccinini committed:
ar71xx: Add Support for the TP-LINK CPE510 V3.0 Device

Based on the work of Andrew Cameron <apcameron@softhome.net> at:
https://github.com/aredn/aredn_ar71xx/commit/230eddf8bbd54bbe3dc99c5dfa5f2d5b1ee1734d

spiccinini committed:
remove files/etc/lime-defaults-factory

As we are currently using this buildroot to build not only for the
LibreRouter but also for other devices then we should not hardcode
functionality that is only intended for the LibreRouter. Lets fix this
by other means.

spiccinini committed:
uci-defaults to add librerouter feeds

The librerouter feed is intended to provide binary release dependent
packages like kernel modules.

spiccinini committed:
Revert "enable busybox CROND_SPECIAL_TIMES @reboot, @hourly, etc"

This reverts commit dfce6c57901dc9b9894ef55f3dcebf6ea6e57978.

spiccinini committed:
uhttpd: fix script-timeout not working if headers sent

spiccinini committed:
add librerouter-hw package with an initial tool to validate hardware

spiccinini committed:
Merge pull request #21 from spiccinini/librerouter-hw-checks

add librerouter-hw package with an initial tool to validate hardware

spiccinini committed:
git: do not ignore .orig nor .rej

I lost a lot of time because I was not seeing these files and they were
getting inside the image /etc/board.d/02_network.orig ... leading to
very unexpected results...

spiccinini committed:
select bandwidth-test package

spiccinini committed:
select deferable-reboot

spiccinini committed:
select lime-report

spiccinini committed:
reorder package shared-state-nodes_and_links

spiccinini committed:
Revert "remove files/etc/lime-defaults-factory"

This reverts commit d5d9198859c3545a09be2f0d384f3b301a5af24b.

spiccinini committed:
om-watchdog is automatically selected when building librerouter

spiccinini committed:
fix LibreMesh git version for release

spiccinini committed:
update order of 5GHz channel for latest libremesh

spiccinini committed:
update lime-packages

spiccinini committed:
move wifi config from lime-defaults-factory to lime-community

luandro committed:
Merge branch 'librerouter-18.06' of https://github.com/LibreRouterOrg/openwrt into librerouter-18.06

nicopace commented on files/etc/config/lime-defaults-factory:
```diff
@@ -0,0 +1,63 @@
+# Beware this file is NOT supposed to be edited by the end user, modify /etc/config/lime instead
```
do not see why this changes are needed in this PR, please remove

nicopace commented on package/utils/busybox/config/miscutils/Config.in:
```diff
@@ -157,6 +157,22 @@ config BUSYBOX_CONFIG_FEATURE_CROND_CALL_SENDMAIL
 	help
 	  Command output will be sent to corresponding user via email.
 
+config BUSYBOX_CONFIG_FEATURE_CROND_SPECIAL_TIMES
```
do not see why this changes are needed in this PR, please remove. thanks!
